packages/core/src/messagingProtocols/proteus/ProteusService/CryptoClient/CoreCryptoWrapper/CoreCryptoWrapper.ts

```diff
@@ -177,8 +177,31 @@ export class CoreCryptoWrapper implements CryptoClient {
     return this.coreCrypto.proteusCryptoboxMigrate(dbName);
   }
 
+  async handleSafeWipe() {
+    //See https://github.com/wireapp/wire-web-packages/pull/4972 for more details
+    const wipeWhenReady = async (onReady: () => void) => {
+      const strongRefCount = this.coreCrypto.strongRefCount();
+      const isCoreCryptoReady = strongRefCount <= 1;
+      if (isCoreCryptoReady) {
+        await this.coreCrypto.wipe();
+        return onReady();
+      }
+    };
+
+    return new Promise<void>(async resolve => {
+      await wipeWhenReady(resolve);
+
+      const intervalId = setInterval(async () => {
+        await wipeWhenReady(() => {
+          clearInterval(intervalId);
+          return resolve();
+        });
+      }, 500);
+    });
+  }
+
   async wipe() {
     await this.config.onWipe();
-    return this.coreCrypto.wipe();
+    return this.handleSafeWipe();
```

Comment:
How about an API like `this.onReady(() => this.coreCrypto.wipe())`

And implementing this generic `onReady` and avoid mixing wiping concerns with readiness concerns. 
WDYT?